arcgis-ios-sdk-samples/Layers/Browse WFS layers/BrowseWFSLayers.storyboard

```diff
@@ -0,0 +1,147 @@
+                        <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
+                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                        <color key="backgroundColor" cocoaTouchSystemColor="groupTableViewBackgroundColor"/>
+                        <prototypes>
+                            <tableViewCell clipsSubviews="YES" contentMode="scaleToFill" preservesSuperviewLayoutMargins="YES" selectionStyle="none" indentationWidth="10" reuseIdentifier="LayerCell" id="3IS-kH-w54">
+                                <rect key="frame" x="0.0" y="55.5" width="375" height="44"/>
+                                <autoresizingMask key="autoresizingMask"/>
+                                <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" preservesSuperviewLayoutMargins="YES" insetsLayoutMarginsFromSafeArea="NO" tableViewCell="3IS-kH-w54" id="mfQ-oX-Lx8">
+                                    <rect key="frame" x="0.0" y="0.0" width="375" height="43.5"/>
+                                    <autoresizingMask key="autoresizingMask"/>
+                                </tableViewCellContentView>
+                            </tableViewCell>
+                            <tableViewCell clipsSubviews="YES" contentMode="scaleToFill" preservesSuperviewLayoutMargins="YES" selectionStyle="none" indentationWidth="10" reuseIdentifier="SettingsCell" id="Zyv-ID-psG" customClass="SettingsTableViewCell" customModule="ArcGIS_Runtime_SDK_Samples" customModuleProvider="target">
+                                <rect key="frame" x="0.0" y="99.5" width="375" height="44"/>
+                                <autoresizingMask key="autoresizingMask"/>
+                                <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" preservesSuperviewLayoutMargins="YES" insetsLayoutMarginsFromSafeArea="NO" tableViewCell="Zyv-ID-psG" id="0eh-SC-W3k">
+                                    <rect key="frame" x="0.0" y="0.0" width="375" height="43.5"/>
+                                    <autoresizingMask key="autoresizingMask"/>
+                                    <subviews>
+                                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Swap" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="oo0-DJ-aZi">
+                                            <rect key="frame" x="16" y="11.5" width="42" height="20.5"/>
+                                            <constraints>
+                                                <constraint firstAttribute="width" relation="greaterThanOrEqual" constant="42" id="TiR-N3-rMk"/>
```

Comment:
Neither of these width constraints are necessary since labels have an intrinsic width based on their text. Please remove them. I would suggest adding a horizontal spacing constraint from the trailing edge of the label to the leading edge of the switch, greater than or equal to the minimum distance you want between the two (you can delete the constant value to use the standard spacing if you want).

Reply:
If I remove the width constraint, I don't see the text. I think it's fine as-is, most of our other samples have width for labels too. 

The horizontal spacing between the two doesn't appear necessary too. In fact, if I add it, the switch moves to the right, and gets a little bit cut in the view

Reply:
> If I remove the width constraint, I don't see the text.

In the storyboard canvas or in the view at runtime? I was able to remove the constraint and still see the text in both places.

> most of our other samples have width for labels 

That's unfortunate as the text width can change based on the system font, the content size category (if using an adaptive font), etc. With Auto Layout, it is best to let views with intrinsic content size size themselves rather than constraining them to an exact width or height. As Apple recommends in their [Rules of Thumb](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/AutolayoutPG/WorkingwithConstraintsinInterfaceBuidler.html#//apple_ref/doc/uid/TP40010853-CH10-SW7) when Working with Constraints in Interface Builder, "Avoid giving views a fixed height or width."

> The horizontal spacing between the two doesn't appear necessary

It isn't strictly necessary, but it is a good idea. It would prevent the label from expanding into the switch's territory. That's not a concern for this sample because the label's text is so short, but samples do change over time and the label's text may not always be the same. In addition, often new samples borrow from the implementation of existing samples. This cell could be copied for use in another sample, which has been done before.

> the switch moves to the right, and gets a little bit cut in the view

Did you add a fixed constraint rather than a greater than or equal to constraint?

I'm not trying to cause you more work for nothing. Another rule of thumb from Apple is to "Create constraints between a view and its nearest neighbor." I've found that following those rules as a habit can bring unforeseen benefits.

Reply:
Removing the width did not display the text of the label in runtime because the horizontal spacing (it was greater than or equal to) was large. Once I reduced that, the label displays fine. Could you take another look?

However, do note that having horizontal spacing in this case would again place a restriction. For eg, if the label were to be copied over to a new sample, it would still cause issues if the text of the newer label is longer because of incorrect horizontal spacing. Samples as they are today are mostly designed and coded for the specific workflow/data. So it's not all bad to have constraints specific to the current sample instead of generic ones as long as the expectation is to modify them correctly when copied over. As for programming practices, it's something the users look for consistency and I agree we should follow patterns and best practices.